[PORT] Grab ADT

mark-ads opened the pull request:
## Описание PR

Портирование механики граба.
Делал по ПР от FaDeOkno.
https://github.com/AdventureTimeSS14/space_station_ADT/pull/1051

## Почему / Баланс

Планомерный перенос контента. 

## Техническая информация

Механика GrabProtection взята с ПР Ratyyy.
https://github.com/AdventureTimeSS14/space_station_ADT/pull/1897

Данные файлы еще не изменялись, так как их еще не перенесли:

Resources/Prototypes/ADT/Entities/Mobs/Species/ipc.yml 
Resources/Prototypes/ADT/Entities/Mobs/Species/kobolt.yml 
Resources/Prototypes/ADT/Entities/Mobs/Species/shadekin.yml

Так же, нет порта ClothingGrantComponent, в связи с чем пока невозможно добавить GrabProtection в следущие файлы:

Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml
Resources/Prototypes/ADT/Heretic/Entities/Clothing/OuterClothing/armor.yml
Resources/Prototypes/ADT/Entities/Clothing/OuterClothing/modsuits.yml

## Чейнджлог

:cl: дс: @kilobana
- add: Схватите своего врага грабом.
- add: Ходят слухи, что некоторые существа научились выходить из граба до его начала.

mark-ads committed:
Перенос механики граба АДТ.
Важно: данные файлы еще не изменялись, так как расы еще не перенесли.
Resources/Prototypes/ADT/Entities/Mobs/Species/ipc.yml
Resources/Prototypes/ADT/Entities/Mobs/Species/kobolt.yml
Resources/Prototypes/ADT/Entities/Mobs/Species/shadekin.yml

Unlumy commented:
Привет, насколько я помню у нас вносились некоторые изменения в граб.
Надо бы глянуть есть ли там что-то кроме изменений для БИ.

Unlumy commented:
Но в целом, @mark-ads вроде проблем не вижу, просто надо проверить разок на наличие каких-то изменений с других пуллов.

mark-ads committed:
Скопировано с ПР Ratyyy
https://github.com/AdventureTimeSS14/space_station_ADT/pull/1897

Добавлена система GrabProtection.
Некоторым монстрам, а так же базовому классу мехов добавлен GrabProtection.
Не добавлена часть изменений Prototypes/ADT в связи отстутстием портирования.

mark-ads commented:
Возможно, еще остались пулл реквесты связанные с грабом, но ими я займусь не раньше среды.
